lib/services/notification-api-service.js

```diff
@@ -109,6 +109,147 @@ function NotificationApiServiceFactory(
+                return updateDeviceInfo(data);
+            })
+            .then(function(data){
+                amqpMessage = data;
+                return eventsProtocol.publishExternalEvent(data)
+                    .then(function () {
+                        return amqpMessage;
+                    });
+            });
+    };
+
+    function getAlert (req) {
+        var amqpMessage = {};
+        var ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress ||
+            req.socket.remoteAddress || req.connection.socket.remoteAddress;
+        logger.info("Alert was received from the node with IP: " + ip);
+        var alert = _.defaults(req.swagger.query || {},req.query || {}, req.body || {});
+        if(typeof(alert) === 'object' &&  Object.keys(alert).length > 0) {
+            //Format the amqp message by adopting the rackHD notification standard
+            amqpMessage.type = "node";
+            amqpMessage.action = "alerts";
+            amqpMessage.data = alert;
+            amqpMessage.data.sourceIpAddress = ip;
+            if (alert.hasOwnProperty("Severity")) {
+                amqpMessage.severity = alert.Severity.toLowerCase();
+            }
+
+            return amqpMessage;
+        }
+        else{
+            //Unexpected decoded "x-www-forum-urlencoded" data
+            throw new Errors.BadRequestError(
+                'Unexpected alert data'
+            );
+        }
+    }
+
+    function updateBmcInfo(amqpMessage) {
+        var ip = amqpMessage.data.sourceIpAddress;
+
+        return waterline.catalogs.findMostRecent({ "data.IP Addresss": ip, source: "bmc"})
+            .then(function(bmcCatalog) {
+                if (bmcCatalog !== undefined && bmcCatalog.hasOwnProperty("node") && bmcCatalog.node !== "") {// jshint ignore:line
+                    return extractBmcInfo(amqpMessage,bmcCatalog);
+                }
+                if (bmcCatalog === undefined){
+                    //Sometimes the IP in the bmc catalog is 0.0.0.0.
+                    //In this case will try to get the IP fromm the obms collection
+                    return waterline.obms.findOne({"config.host": ip})
+                        .then(function(node) {
+                            if (node !== undefined) {
+                                return waterline.catalogs.findMostRecent({"node": node.node, source: "bmc"})// jshint ignore:line
+                                    .then(function (bmcCatalog) {
+                                        if (bmcCatalog !== undefined && bmcCatalog.hasOwnProperty("node") && bmcCatalog.node !== "") { // jshint ignore:line
+                                            return extractBmcInfo(amqpMessage, bmcCatalog);
+                                        }
+                                    });
+                            }else{
+                                //unrecognized node
+                                throw new Errors.BadRequestError("unrecognized node");
```

Comment:
@yyscamper  it actually means that the node wasn't found neither by looking the obm collection or the bmc catalog.  I will leave the error description as it is because it is accurate.